packages/expo-manifests/ios/EXManifests/EXManifestsManifest.swift

```diff
@@ -0,0 +1,359 @@
+    return self.expoClientConfigRootObject().let { it in
+      EXManifestsManifest.string(fromManifest: it, atPaths: [
+        ["ios", "splash", "backgroundColor"],
+        ["splash", "backgroundColor"]
+      ])
+    }
+  }
+
+  public func iosSplashImageUrl() -> String? {
+    return self.expoClientConfigRootObject().let { it in
+      EXManifestsManifest.string(fromManifest: it, atPaths: [
+        UIDevice.current.userInterfaceIdiom == UIUserInterfaceIdiom.pad
+          ? ["ios", "splash", "tabletImageUrl"] : [],
+        ["ios", "splash", "imageUrl"],
+        ["splash", "imageUrl"]
+      ])
+    }
+  }
+
+  public func iosSplashImageResizeMode() -> String? {
+    return self.expoClientConfigRootObject().let { it in
+      EXManifestsManifest.string(fromManifest: it, atPaths: [
+        ["ios", "splash", "resizeMode"],
+        ["splash", "resizeMode"]
+      ])
+    }
+  }
+
+  public func iosGoogleServicesFile() -> String? {
+    return self.iosConfig()?.optionalValue(forKey: "googleServicesFile")
+  }
+
+  public func supportsRTL() -> Bool {
+    guard let expoClientConfigRootObject = self.expoClientConfigRootObject(),
+      let extra: [String: Any]? = expoClientConfigRootObject.optionalValue(forKey: "extra"),
+      let supportsRTL: Bool = extra?.optionalValue(forKey: "supportsRTL") else {
+      return false
+    }
+
+    return supportsRTL
+  }
+
+  public func jsEngine() -> String {
+    let jsEngine = self.expoClientConfigRootObject().let { it in
+      EXManifestsManifest.string(fromManifest: it, atPaths: [
+        ["ios", "jsEngine"],
+        ["jsEngine"]
+      ])
+    }
+
+    guard let jsEngine = jsEngine else {
+      let sdkMajorVersion = self.sdkMajorVersion()
+      if sdkMajorVersion > 0 && sdkMajorVersion < 48 {
+        return "jsc"
+      } else {
+        return "hermes"
+      }
+    }
+    return jsEngine
+  }
+
+  private func sdkMajorVersion() -> Int {
+    let sdkVersion = self.sdkVersion()
+    let components = sdkVersion?.components(separatedBy: ".")
+    guard let components = components else {
+      return 0
+    }
+    if components.count == 3 {
+      let ret = Int(components[0])
+      guard let ret = ret else {
+        return 0
+      }
+      return ret
+    }
+    return 0
+  }
+
+  private static func string(fromManifest: [String: Any], atPaths: [[String]]) -> String? {
+    for path in atPaths {
+      if let result = self.string(fromManifest: fromManifest, atPath: path) {
+        return result
+      }
+    }
+    return nil
+  }
+
+  private static func string(fromManifest: [String: Any], atPath: [String]) -> String? {
+    var json = fromManifest
+    for i in 0..<atPath.count {
+      let isLastKey = i == atPath.count - 1
+      let key = atPath[i]
+      let value = json[key]
+      if isLastKey && value is String {
+        // Type check above preempts this force_cast
+        // swiftlint:disable:next force_cast
+        return value as! String?
```

Comment:
Then force casting is not needed
```suggestion
        return value
```